alchemist-graphql/src/commonMain/resources/graphql/schema.graphqls

```diff
@@ -0,0 +1,185 @@
+  "Returns the value of the layer at the given position"
+  getValue(p: PositionInput!): String!
+}
+
+"A molecule with an associated name"
+type MoleculeSurrogate {
+  name: String!
+}
+
+"A pair Molecule-Concentration"
+type MoleculeToConcentrationEntry {
+  concentration: String!
+  molecule: MoleculeSurrogate!
+}
+
+"A map storing molecules and their concentrations"
+type MoleculeToConcentrationMap {
+  "The list of pairs Molecule-Concentration"
+  entries: [MoleculeToConcentrationEntry!]!
+  size: Int!
+}
+
+type Mutation {
+  "Pause the simulation"
+  pause: String!
+  "Play the simulation"
+  play: String!
+  "Terminate the simulation"
+  terminate: String!
+}
+
+"A neighborhood of nodes"
+type NeighborhoodSurrogate {
+  "The central node of this neighborhood"
+  getCenter: NodeSurrogate!
+  "The list of the neighbors"
+  getNeighbors: [NodeSurrogate!]!
+  "Whether this neighborhood is empty"
+  isEmpty: Boolean!
+  size: Int!
+}
+
+"A node in the environment"
+type NodeSurrogate {
+  "Whether the given molecule is present"
+  contains(molecule: MoleculeInput!): Boolean!
+  "List of Molecule-Concentration pairs"
+  contents: MoleculeToConcentrationMap!
+  "The concentration of the given molecule"
+  getConcentration(molecule: MoleculeInput!): String
+  id: Int!
+  moleculeCount: Int!
+  "The list of properties/capabilities of this node as a string"
+  properties: [String!]!
+  "The list of reactions of this node"
+  reactions: [ReactionSurrogate!]!
+}
+
+"The pair NodeId-Position"
+type NodeToPosEntry {
+  id: Int!
+  position: PositionSurrogate!
+}
+
+"Map of nodes' ids and their position"
+type NodeToPosMap {
+  "The list of pairs NodeId-Position"
+  entries: [NodeToPosEntry!]!
+  size: Int!
+}
+
+"A position in a two dimensional space"
+type Position2DSurrogate implements PositionSurrogate {
+  coordinates: [Float!]!
+  dimensions: Int!
+  x: Float!
+  y: Float!
+}
+
+type Query {
+  "The actual state of the environment"
+  environment: EnvironmentSurrogate!
+  "The neighborhood which the given node is the center."
+  neighborhood(nodeId: Int!): NeighborhoodSurrogate!
+  "The position in space of the given node."
+  nodePosition(nodeId: Int!): PositionSurrogate
+  "The status of the simulation"
+  simulationStatus: String!
+}
+
+"A generic reaction"
+type ReactionSurrogate {
+  inputContext: Context!
+  node: NodeSurrogate!
+  outputContext: Context!
+}
+
+type Subscription {
+  "The simulation's environment"
+  environment: EnvironmentSurrogate!
+  "The neighborhood of a node in the simulation's environment"
+  neighborhood(nodeId: Int!): NeighborhoodSurrogate!
+  "A node in the simulation's environment"
+  node(nodeId: Int!): NodeSurrogate!
+}
+
+enum Context {
+  GLOBAL
+  LOCAL
+  NEIGHBORHOOD
+}
+
+"A molecule with an associated name, used as input object"
+input MoleculeInput {
+  name: String!
+}
+
+"Position in space, used as input parameter"
+input PositionInput {
+  coordinates: [Float!]!
+  dimensions: Int!
+}
```

Comment:
This looks generated... Is it?

Reply:
This is downloaded from the server endpoint while it is up and running and it is needed at compile time in for the client.

Reply:
So it's generated, is it correct? If so, we could write a gradle task that generates these files (it should be there already, actually), and then alter the classpath of the client to include these files

Reply:
Yes it is generated, but it can be downloaded only when the server is up, and the client requires the schema to be present when it is being compiled.

There is a task called `graphqlGenerateSDL` that generate the schema at compile time, but it makes use of `NoopSchemaGeneratorHooks` which, in a few words, cannot generate the schema in our case because it is not designed to support Subscriptions that use Kotlin `Flow`s. 
For those operation we need `FlowSubscriptionSchemaGeneratorHooks` which is used inside the server, and so the schema is properly generated at runtime, but at compile time that set of hooks are not available inside the GraphQL Kotlin gradle plugin.

I have searched in the [documentation](https://opensource.expediagroup.com/graphql-kotlin/docs/plugins/gradle-plugin-tasks#graphqlgeneratesdl) how to add a set of custom hooks that can be used by the gradle plugin, but it is quite complex to do (see [the guide](https://opensource.expediagroup.com/graphql-kotlin/docs/plugins/hooks-provider#creating-custom-hooks-service-provider)).
In case, I could raise an issue at GraphQL Kotlin, pointing out that they could provide `FlowSubscriptionSchemaGeneratorHooks` support inside the gradle plugin, as they do with federated schemas.
Otherwise I can try to build a custom one that exposes the flow subscription hooks.

Reply:
Can we start the server and then grab those files before the compilation of the client?
The server is in a separate project, isn't it? We could have a task that launches it for a snap, making it generate the files, and then we instruct the compiler to add them to the compile classpath.

Not a trivial Gradle exercise, but I believe it's doable. Maybe we can try to do it together

Reply:
Should be done, right @AngeloFilaseta and @S-furi ?

Reply:
Yes to me this is fixed!